runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -222,6 +227,45 @@ DataflowPipelineJob run(Pipeline pipeline, DataflowRunner runner) {
     return Optional.<Boolean>absent();
   }
 
+  /**
+   * Check if all data watermark value for each step of the pipeline reach to max.
+   */
+  boolean atMaxWatermark(DataflowPipelineJob job) {
+    JobMetrics metrics;
+    try {
+      metrics = options.getDataflowClient().projects().jobs()
+          .getMetrics(job.getProjectId(), job.getJobId()).execute();
+    } catch (IOException e) {
+      LOG.error("Failed to retrieve job metrics. {}", e);
```

Comment:
1. change to warn
2. Clarify exception is ignored, something like:
LOG.warn("Ignore the exception {} while retrieving job metrics", e);